### PR TITLE
In `parse`, return Invalid Date if there is any remaining non-whitespace input (closes #819)

### LINKED ISSUE #819: Options for strict parsing?
I'm using `date-fns` (latest 2.0 alpha) to validate that certain datetime strings are in a specific ISO8601 compliant format. It's critically important that they are in the exact format I'm testing for so that certain invariants hold.

However, this parses as valid:

```
> dateFns.parse("2012-12-12T12:30", "YYYY-MM-dd", new Date())
2012-12-11T22:00:00.000Z
```

and even this:

```
> dateFns.parse("2012-12-12asfasfasfT12:30", "YYYY-MM-dd", new Date())
2012-12-11T22:00:00.000Z
```

I propose the parser should check if there's any remaining input after parsing, as long as a format was provided. Best-effort parsing is okay for UI use, but for critical validation tasks this is unwanted behaviour.

There should also be an option for disabling the `Date` constructor fallback, for the same reasons.

### PR DESCRIPTION
**Pull request by leshakoss**
See #819 for more details

### CODE REVIEW

File: src/parse/_lib/parsers/index.js [hunk 1]
```diff
@@ -517,7 +517,6 @@ var parsers = {
       }
     },
     validate: function (date, value, options) {
-      console.log({value})
       return value >= 0 && value <= 11
```
Comment (kossnocorp) [inline src/parse/_lib/parsers/index.js]: Hah

Comment (curry684) [reply src/parse/_lib/parsers/index.js]: That line just cost me an hour because it was crashing my webpack builds in CI   😢

Comment (leshakoss) [reply src/parse/_lib/parsers/index.js]: @curry684 sorry :[ It's removed in v2.0.0-alpha.16

Comment (curry684) [reply src/parse/_lib/parsers/index.js]: Hehe yeah already upgraded, they're succeeding again. My own fault for using alpha  ;)

**Pull request merged**